docs/machine-learning/how-to-guides/prepare-data-ml-net.md

```diff
@@ -0,0 +1,331 @@
+---
+title: "How-To: Prepare Data with ML.NET"
+description: Learn how to use transforms in ML.NET to manipulate and prepare data for additional processing or model building.
+ms.date: 04/05/2019
+ms.custom: mvc,how-to
+#Customer intent: As a developer, I want to know how I can transform and prepare data with ML.NET using transforms
+---
+
+# How-To: Prepare Data with ML.NET
+
+This how-to shows how to use transformers in ML.NET to prepare data for additional processing or building a model.
+
+## Replace missing values
+
+Missing values are a common occurrence in datasets. One approach to dealing with missing values is to replace them with a placeholder or another meaningful value such as the mean value in the data. 
+
+Using the following input data which is loaded into an [`IDataView`](xref:Microsoft.ML.IDataView):
+
+```csharp
+HomeData[] homeDataList = new HomeData[]
+{
+    new HomeData
+    {
+        NumberOfBedrooms=1f,
+        Price=100000f
+    },
+    new HomeData
+    {
+        NumberOfBedrooms=2f,
+        Price=300000f
+    },
+    new HomeData
+    {
+        NumberOfBedrooms=6f,
+        Price=float.NaN
+    }
+};
+```
+
+Notice that the last element in our list has a missing value for `Price`. To replace the missing values in the `Price` column, use the [`ReplaceMissingValue`](xref:Microsoft.ML.ExtensionsCatalog.ReplaceMissingValues*) method to fill in that missing values.
+
+> [!IMPORTANT]
+> [`ReplaceMissingValue`](xref:Microsoft.ML.ExtensionsCatalog.ReplaceMissingValues*) only works with numerical data.
+
+```csharp
+// Define replacement estimator
+var replacementEstimator = mlContext.Transforms.ReplaceMissingValues("Price", replacementMode: MissingValueReplacingEstimator.ReplacementMode.Mean);
+
+// Build transformer to perform operations
+ITransformer replacementTransformer = replacementEstimator.Fit(data);
+
+// Transform data
+IDataView transformedData = replacementTransformer.Transform(data);
+```
+
+ML.NET supports various [replacement modes](xref:Microsoft.ML.Transforms.MissingValueReplacingEstimator.ReplacementMode). The sample above uses the `Mean` replacement mode which will fill in the missing value with the average value for that column. The result of the replacement fills in the `Price` property for the last element in our data with 200000 since it's the average of 100000 and 300000. 
```

Comment:
"the average value for that column" -> "that column's average value"
"The result of the replacement" -> "The replacement's result"